extensions/ql-vscode/src/local-databases-ui.ts

```diff
@@ -496,26 +405,49 @@ export class DatabaseUI extends DisposableObject {
         )`Failed to choose and set database: ${getErrorMessage(e)}`,
       );
     }
-  };
+  }
+
+  private async handleChooseDatabaseArchive(): Promise<void> {
+    return withProgress(
+      async (progress, token) => {
+        await this.chooseDatabaseArchive(progress, token);
+      },
+      {
+        title: "Adding database from archive",
+      },
+    );
+  }
 
-  handleChooseDatabaseInternet = async (
+  public async chooseDatabaseInternet(
     progress: ProgressCallback,
     token: CancellationToken,
-  ): Promise<DatabaseItem | undefined> => {
+  ): Promise<DatabaseItem | undefined> {
     return await promptImportInternetDatabase(
       this.databaseManager,
       this.storagePath,
       progress,
       token,
       this.queryServer?.cliServer,
     );
-  };
+  }
 
-  handleChooseDatabaseGithub = async (
-    credentials: Credentials | undefined,
+  private async handleChooseDatabaseInternet(): Promise<void> {
```

Comment:
Yes, exactly. We will convert the commands in `extension.ts` to use typed commands as well, including using `withProgress` instead of `commandRunnerWithProgress`. The reason for using two different methods is because for some of these the text that is shown in the progress notification is different, so we can't re-use the same method. I think it's also a somewhat clearer separation if we have separate methods which can be called publicly vs handling commands privately.